### ISSUE TITLE
[BUG] Embedding combo alias fails for llama-cpp embedding-only model and combo test uses chat/completions

### ISSUE THREAD

**Issue by kserge1t** [diegosouzapw/OmniRoute#1260]
### OmniRoute Version

v3.6.5

### Installation Method

Docker / Docker Compose

### Operating System

Linux

### OS Version

Ubuntu 24.04.4 LTS 

### Node.js Version

Docker

### Provider(s) Involved

Llama Cpp (Custom local OpenAI Compatible provider)

### Model(s) Involved

llama-cpp/embeddinggemma-300M-Q8_0

### Client Tool

OpenClaw, OmniRoute dashboard, cURL

### Description

Embedding combo aliases do not appear to work correctly when one leg is a llama.cpp embedding-only model.

I created a combo named `embed-embeddinggemma` containing:
- a llama.cpp local embedding model (`llama-cpp/embeddinggemma-300M-Q8_0`)

The underlying embedding models work correctly through OmniRoute on `/v1/embeddings` (when used directly, not via combo).

However:
1. The combo "test" action appears to probe the llama.cpp leg via `/v1/chat/completions` instead of `/v1/embeddings`
2. The raw embeddings API rejects the combo alias `embed-embeddinggemma` instead of resolving it as an embedding-capable model

This makes embedding combos look broken/unusable even though the underlying embedding provider/model is healthy.

### Steps to Reproduce

1. Configure a working llama.cpp embedding model in OmniRoute:
   `llama-cpp/embeddinggemma-300M-Q8_0`

2. Confirm the Actual model works:
   send `POST /v1/embeddings` with `model=llama-cpp/embeddinggemma-300M-Q8_0`

3. Create an embedding combo alias named `embed-embeddinggemma`
   containing:
   - LM Studio embedding model (lan-hosted GPU-enabled)
   - llama.cpp embedding model (localhost CPU-only)

4. In the OmniRoute UI, click the combo test action for `embed-embeddinggemma`

5. Also call `POST /v1/embeddings` with:
   `model=embed-embeddinggemma`

### Expected Behavior

- The combo test should validate embedding combos using the embeddings path, not chat/completions
- `embed-embeddinggemma` should behave as a valid embedding model alias on `/v1/embeddings`
- If one provider in the combo is healthy, the combo should either:
  - succeed using the configured routing/fallback behavior, or
  - fail with an embeddings-specific error that reflects real embedding behavior

### Actual Behavior

- The combo test appears to send the llama.cpp embedding-only model through `/v1/chat/completions`
- The llama.cpp leg fails with:
  `500: the current context does not logits computation`

- Raw embeddings request with the combo alias fails with:
  `400 Invalid embedding model: embed-embeddinggemma. Use format: provider/model`
- Meanwhile, the actual underlying embedding model still works correctly:
  `POST /v1/embeddings` with `model=llama-cpp/embeddinggemma-300M-Q8_0` returns HTTP 200

### Test Impact

Needs a new integration test

### Error Logs / Output

```shell
Actual model works:
POST /v1/embeddings
model=llama-cpp/embeddinggemma-300M-Q8_0
-> HTTP 200
-> embedding length 768

Combo alias rejected on embeddings endpoint:
POST /v1/embeddings
model=embed-embeddinggemma
-> HTTP 400
-> Invalid embedding model: embed-embeddinggemma. Use format: provider/model

Combo test / logs show wrong API path for llama.cpp embedding-only leg:
POST /v1/chat/completions
target provider/model: openai-compatible-chat-.../embeddinggemma-300M-Q8_0
-> 500: the current context does not logits computation
```

### Additional Context

Environment details:
- OmniRoute running in Docker (`diegosouzapw/omniroute:latest`)
- llama.cpp backend is local (localhost) CPU-only
- LM Studio backend is local (LAN) GPU

Actual OmniRoute model ID currently working (when used directly, not via Combo):
- `llama-cpp/embeddinggemma-300M-Q8_0`

This looks like a feature gap or bug in combo alias resolution for embeddings, not a bad model setup:
- Actual embedding model works
- combo alias does not resolve on `/v1/embeddings`
- combo tester appears chat-centric and incompatible with embedding-only models

### Validation Plan

1. Create an embedding combo alias containing at least one embedding-only llama.cpp model
2. Verify combo test uses `/v1/embeddings` instead of `/v1/chat/completions`
3. Verify `POST /v1/embeddings` with `model=embed-embeddinggemma` returns HTTP 200
4. Verify returned embedding shape matches the selected/resolved provider model
5. Add regression coverage for:
   - embedding combo alias resolution
   - embedding-only provider legs
   - combo test path selection based on model capability

**Comment by oyi77** [comment]
## Fix Implemented

I've pushed a fix for this issue in PR #1281 (commit `f63e7b59`).

### Root Cause (2 sub-problems)

1. **`/v1/embeddings` rejects combo names**: `parseEmbeddingModel()` expects `provider/model` format, so passing a combo name like `embed-embeddinggemma` fails with "Invalid embedding model".

2. **Combo test uses chat/completions for embedding combos**: `buildComboTestRequestBody()` always builds a chat request, and the test endpoint always targets `/v1/chat/completions` — meaningless for embedding models.

### Fix

**Sub-problem 1** (`src/app/api/v1/embeddings/route.ts`):
Before calling `parseEmbeddingModel()`, the route now checks if `body.model` is a combo name via `getComboByName()`. If resolved, it uses `resolveNestedComboTargets()` to get the first target's `provider/model` string. Non-combo model strings pass through unchanged.

**Sub-problem 2** (`src/lib/combos/testHealth.ts` + `src/app/api/combos/test/route.ts`):
- Added `buildComboTestRequestBodyEmbedding()` — sends `{ model, input: "..." }` instead of chat messages
- Added `isEmbeddingModel()` — detects embedding models by pattern (embed, e5, bge, text-embedding, etc.)
- Combo test route now detects embedding combos and uses `/v1/embeddings` endpoint with embedding-specific response validation (checks for `data` array with vectors instead of `choices` with text)

**Comment by diegosouzapw** [comment]
Fixed in release/v3.6.6. The fix will be included in the next release.
